### PR TITLE
test: enhance the network isolated e2e test to avoid cached binary

### PR DESCRIPTION
**Pull request by fseldow**
**What type of PR is this?**

/kind test

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:
Current airgap e2e does not catch the oras pull logic when binary is already cached
Thus use custom vhd

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/config/vhd.go [hunk 1]
```diff
@@ -61,6 +61,13 @@ var (
 		Arch:    "amd64",
 		Version: "1.1704411049.2812",
 	}
+
+	VHDUbuntu2204Gen2ContainerdAirgapped = &Image{
+		Name:    "2204gen2containerd",
+		OS:      "ubuntu",
+		Arch:    "amd64",
+		Version: "1.1725544348.4398", // without cached kubelet, kubectl
```
Comment (cameronmeissner) [inline e2e/config/vhd.go:69]: please make sure to add a resource lock around this particular SIG image version

File: e2e/scenario_test.go [hunk 2]
```diff
@@ -652,7 +652,33 @@ func Test_ubuntu2204AirGap(t *testing.T) {
 		},
 		Config: Config{
 			Cluster: ClusterKubenetAirgap,
-			VHD:     config.VHDUbuntu2204Gen2Containerd,
+			VHD:     config.VHDUbuntu2204Gen2ContainerdAirgapped,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-22.04-gen2"
+				nbc.AgentPoolProfile.Distro = "aks-ubuntu-containerd-22.04-gen2"
+
+				// TODO(xinhl): define below in the cluster config instead of mutate bootstrapConfig
```
Comment (cameronmeissner) [inline e2e/scenario_test.go:660]: mutating bootstrap configuration is generally how we define scenarios, I'd avoid changing the base cluster config if possible

File: e2e/scenario_test.go [hunk 3]
```diff
@@ -652,7 +652,33 @@ func Test_ubuntu2204AirGap(t *testing.T) {
...
+						Enabled:                 true,
+						ContainerRegistryServer: "mcr.microsoft.com",
+					},
+				}
+			},
+		},
+	})
+}
+
+func Test_Ubuntu2204Gen2ContainerdAirgapped_NoK8s(t *testing.T) {
```
Comment (cameronmeissner) [inline e2e/scenario_test.go:673]: nit: `Test_Ubuntu2204Gen2ContainerdAirgapped_K8sNotCached`

Comment (fseldow) [reply e2e/scenario_test.go]: updated

File: e2e/scenario_test.go (hunk 2 above)
Comment (fseldow) [reply e2e/scenario_test.go]: remove the todo

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**